packages/protobuf/src/from-json.ts

```diff
@@ -233,14 +240,54 @@ function readField(
       readMessageField(msg, field, json, opts);
       break;
     case "list":
-      readListField(msg, field, json, opts);
+      readListField(msg.get(field), field, json, opts);
       break;
     case "map":
       readMapField(msg, field, json, opts);
       break;
   }
 }
 
+function readExtension(
+  field: DescExtension,
+  json: JsonValue,
+  opts: JsonReadOptions,
+) {
+  try {
+    switch (field.fieldKind) {
+      case "scalar":
+        return readScalar(field.scalar, json, field.longType, false);
+      case "enum":
+        return readEnum(field.enum, json, opts.ignoreUnknownFields, false);
+      case "message":
+        if (json === null) return tokenNull;
+        const messageValue = create(field.message);
+        readMessage(reflect(field.message, messageValue), json, opts);
+        return messageValue;
+      case "list":
+        if (json === null) return tokenNull;
+        const listValue: unknown[] = [];
+        const list = {
+          add: (...items) => void listValue.push(...items),
+        } satisfies Pick<ReflectList, "add">;
+        if (field.listKind === "message") {
+          list.add = (...items) =>
+            void listValue.push(
+              items.map((v) => (v as ReflectMessage).message),
+            );
+        }
+        readListField(list, field, json, opts);
```

Comment:
We are able to reuse this be tweaking the function a little, I copied the logic in binary, not sure what works better.

Reply:
It's probably also an option to tweak reflectList a bit to accept extensions, but I'm not sure that works better either.

Let me check our test coverage and dig into this a bit.